express-postgres/server.ts

```diff
@@ -0,0 +1,85 @@
+import pg from 'pg';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.get('/api/films', async (req, res, next) => {
+  try {
+    const sql = `
+      select "title",
+             "filmId"
+      from "films"
+      order by "replacementCost" desc;
+    `;
+    const result = await db.query(sql);
+    if (!result) {
+      throw new ClientError(404, 'films not found');
+    }
+    res.send(result);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/film', async (req, res, next) => {
+  try {
+    const { filmId } = req.query;
+    if (filmId === undefined) {
+      throw new ClientError(400, 'filmId is required');
+    }
+    const sql = `
+      select *
+      from "films"
+      where "filmId" = $1;
+    `;
+    const params = [filmId];
+    const result = await db.query(sql, params);
+    const film = result.rows[0];
+    if (!film) {
```

Comment:
This one makes sense, because if there isn't a film with that ID then the client asked for something that doesn't exist.